icicle/appUtils/msm/msm.cu

```diff
@@ -338,59 +340,63 @@ namespace msm {
     // this function computes msm using the bucket method
     template <typename S, typename P, typename A>
     cudaError_t bucket_method_msm(
-      int bitsize,
-      int c,
+      unsigned bitsize,
```

Comment:
Just a style related note: google guide advices to basically use `int` everywhere. I know that we're still not following this rule fully, but we at least tried to do it in higher-level and public functions. Do you think it's a good idea to stick to it?

Reply:
I know someone who is very experienced with CUDA and he uses uint32_t wherever he can:)

Reply:
Well limbs obviously shouldn't be `int`. I'll just quote google guide to be more precise:

> We use int very often, for integers we know are not going to be too big, e.g., loop counters. Use plain old int for such things.
> ...
> You should not use the unsigned integer types such as uint32_t, unless there is a valid reason such as representing a bit pattern rather than a number, or you need defined overflow modulo 2^N. In particular, do not use unsigned types to say a number will never be negative. Instead, use assertions for this.

Reply:
We discussed it in the daily meeting.
I personally prefer unsigned values (and actually uint32_t as Hadar mentioned since it's more portable and can avoid nasty bugs since sizeof(int) is not guaranteed to be 4 in all systems) when the value should not be negative because it tells the reader that the value should not be negative.

Two notes:
(1) google style suggests asserting the values are not negative instead of using unsigned int. I am not sure it is applicable to kernel code.
(2) when using std containers like std::vector, their size() method returns an unsigned value and therefore the compiler will warn about comparison if the loop counter is signed. Iterators are probably the better design pattern in those cases but this is not really the point here.